docs/source/udf/buffer_types.py

```diff
@@ -1,22 +1,30 @@
 from libertem.udf import UDF
 
-class test_buffer(UDF):
+class TestBuffer(UDF):
 
-	def get_results_buffer(self):
+	def get_result_buffers(self):
 		"""
 		Initialize a buffer of different types
 
+		Suppose our dataset has dimension (14, 14, 32, 32),
+		where the first two dimensions represent the navigation
+		dimension and the last two dimensions represent the signal dimension.
+
 		The first buffer "single_buffer" specifies a buffer of
-		dimension (16, 16) with entries of type float32
+		dimension (16, 16) with entries of type float32.
 
 		The second buffer "sig_buffer" specifies a buffer whose
-		first component dimension is increased by 2 from that of
-		signal dimension
+		dimension is initialized to be the same as the signal dimension.
+		With 'extra_shape = (2,)', two additional dimensions are added
+		to the first dimension of the signal dimension, so the resulting
+		dimension for this buffer will be (34, 32)
 
 		The third buffer "nav_buffer" specifies a buffer whose
-		first component dimension is increased by 1 and the second
-		component dimension is increased by 2 from that of navigation
-		dimension
+		dimension is initialized to be the same as the navigation dimension.
+		With 'extra_shape = (1, 2)', one additional dimension is added to the
+		first dimension of the navigation dimension and two additional
+		dimensions are added to the second dimension of the navigation dimension,
+		so the resulting dimension for this buffer will be (15, 16)
```

Comment:
There is still a misunderstanding here - if you specify `extra_shape`, extra dimensions are introduced! In this example, the full shape would be `(14, 14, 1, 2)` - so you can store two `float32` numbers as a result for each frame. The **size** of the existing navigation dimensions stays the same.

(Same applies to the `sig_buffer` paragraph above)

Reply:
Oh got it. Sorry for the misunderstanding! I'll fix it immemdiately